File: app/src/main/java/com/cliniva/enventory/ui/supplies/SuppliesFragment.java
package com.cliniva.enventory.ui.supplies;


import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.cliniva.enventory.R;
import com.cliniva.enventory.ui.base.BaseFragment;

/**
 * A simple {@link Fragment} subclass.
 */
public class SuppliesFragment extends BaseFragment {


    public SuppliesFragment() {
        // Required empty public constructor
    }

    public static SuppliesFragment getInstance(){
        SuppliesFragment fragment = new SuppliesFragment();
        Bundle args = new Bundle();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public int getLayoutRes() {
        return R.layout.fragment_supplies;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
    }
}

File: app/src/main/java/com/amirsons/inventory/ui/fragment/supplies/SuppliesFragment.kt
package com.amirsons.inventory.ui.fragment.supplies


import android.os.Bundle
import android.util.Log
import android.view.View
import android.view.ViewGroup
import com.amirsons.inventory.R
import com.amirsons.inventory.datamanager.model.TransactionHistory
import com.amirsons.inventory.ui.base.BaseFragment
import com.amirsons.inventory.ui.recyclerview.base.BaseRecyclerClickListener
import com.amirsons.inventory.ui.recyclerview.base.BaseRecyclerViewHolder
import com.amirsons.inventory.ui.recyclerview.base.RecyclerViewAdapter
import com.amirsons.inventory.ui.recyclerview.viewholder.TransactionHistoryHolder
import kotlinx.android.synthetic.main.fragment_supplies.*

/**
 * A simple [BaseFragment] subclass.
 */
class SuppliesFragment : BaseFragment(), SuppliesView, BaseRecyclerClickListener<TransactionHistory> {

    private lateinit var mSuppliesListAdapter: RecyclerViewAdapter<TransactionHistory, BaseRecyclerClickListener<TransactionHistory>>
    private lateinit var mSuppliesPresenter: SuppliesPresenter

    override val contentLayout: Int
        get() = R.layout.fragment_supplies

    companion object {

        val instance: SuppliesFragment
            get() {
                val fragment = SuppliesFragment()
                val args = Bundle()
                fragment.arguments = args
                return fragment
            }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mSuppliesPresenter = SuppliesMvp(this)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        // set toolbar name
        setToolbar(view, context.getString(R.string.title_supplies_toolbar), false)

        initRecyclerView()

        mSuppliesPresenter.onLoadList()
    }

    private fun initRecyclerView() {

        mSuppliesListAdapter = object : RecyclerViewAdapter<TransactionHistory, BaseRecyclerClickListener<TransactionHistory>>() {
            override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseRecyclerViewHolder<TransactionHistory, BaseRecyclerClickListener<TransactionHistory>> {
                return TransactionHistoryHolder(inflate(R.layout.item_transaction_history, parent))
            }
        }

        mSuppliesListAdapter.setListener(this)
        rv_supplies_list.adapter = mSuppliesListAdapter
    }

    override fun setListToView(transactionHistoryList: ArrayList<TransactionHistory>) {
        mSuppliesListAdapter.setItems(transactionHistoryList)
        Log.wtf(TAG, "${transactionHistoryList.size}")
    }

    override fun onItemClickListener(item: TransactionHistory, position: Int) {

    }

    override fun onPause() {
        super.onPause()
        mSuppliesPresenter.onRemoveDatabaseListener()
    }
}
